.github/workflows/compare_latest_version.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+import os
+import re
+from pathlib import Path
+from urllib.parse import urljoin
+
+import requests
+from bs4 import BeautifulSoup
+
+SNAPHU_URL = "https://web.stanford.edu/group/radar/softwareandlinks/sw/snaphu/"
+
+
+def fetch_stanford_version_url() -> str:
+    """
+    Find the link to the latest version of "snaphu-vXXX.tar.gz" from the given URL.
+
+    Parameters
+    ----------
+    url : str
+        The URL of the webpage to search for the snaphu tarball.
+
+    Returns
+    -------
+    str
+        The URL to the latest version of snaphu tarball if found, otherwise an empty string.
+
+    """
```

Comment:
* This function returns two things in the nominal case -- not one.
* When there are multiple return values, I prefer to provide names for each return value in the docstring.
* There shouldn't be a blank line at the end of the docstring.

```suggestion
    Returns
    -------
    version : str
        The version string of the latest available SNAPHU version, in '<major>.<minor>.<patch>' format.
    url : str
        The URL of the latest version of the SNAPHU tarball.
    """
```

>`if found, otherwise an empty string.`

Not a fan of this -- let's raise an exception if the function failed to find the latest tarball.